lib/gollum-lib/filter/render.rb

```diff
@@ -30,6 +30,8 @@ def add_editable_header_class(data)
     doc.css('h1,h2,h3,h4,h5,h6').each_with_index do |header, i|
       next if header.content.empty?
       next if header.inner_html.match(PLACEHOLDER_PATTERN)
+      next unless @markup.wiki.allow_editing
```

Comment:
We could also skip the entire `add_editable_header_class` call if `@markup.wiki.allow_editing` is `false`

Reply:
Good suggestion. Thank you!